File: conformance/bundles/14-stateless-steps/squares.steps.py
from varar import steps

# No state factory: these steps are pure, so steps() is called bare
# and handlers get an empty dict as state.
param, stimulus, sensor = steps()


@stimulus("I warm up my mental math")
def _(state):
    pass


@sensor("The square of {int} is {int}.")
def _(state, n, expected):
    return [n, n * n]

File: conformance/bundles/14-stateless-steps/squares.steps.ts
import { steps } from '@varar/varar'

// No state factory: these steps are pure — nothing to arrange, nothing to
// evolve — so steps() is called bare and handlers get an empty state.
const { stimulus, sensor } = steps()

stimulus('I warm up my mental math', () => {})

sensor('The square of {int} is {int}.', (_state, n: number) => [n, n * n])
